File: freecodecamp/daily-code-challenge/2026-03-05.py
def smallest_gap(s):
    last_seen = {}
    min_gap = float('inf')
    result = ""

    for i, char in enumerate(s):
        if char in last_seen:
            gap = i - last_seen[char] - 1

            if gap < min_gap:
                min_gap = gap
                result = s[last_seen[char] + 1:i]

        last_seen[char] = i

    return result

File: freecodecamp/daily-code-challenge/2026-03-05.js
function smallestGap(str) {
  let lastSeen = {};
  let minGap = Infinity;
  let result = "";

  for (let i = 0; i < str.length; i++) {
    let char = str[i];

    if (char in lastSeen) {
      let gap = i - lastSeen[char] - 1;

      if (gap < minGap) {
        minGap = gap;
        result = str.slice(lastSeen[char] + 1, i);
      }
    }

    lastSeen[char] = i;
  }

  return result;
}
